src/pages/workspace/accounting/intacct/import/SageIntacctEditUserDimensionsPage.tsx

```diff
@@ -0,0 +1,136 @@
+import FormProvider from '@components/Form/FormProvider';
+import InputWrapper from '@components/Form/InputWrapper';
+import type {FormInputErrors, FormOnyxValues} from '@components/Form/types';
+import * as Expensicons from '@components/Icon/Expensicons';
+import MenuItem from '@components/MenuItem';
+import OfflineWithFeedback from '@components/OfflineWithFeedback';
+import TextInput from '@components/TextInput';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {clearSageIntacctUserDimensionErrorField, editSageIntacctUserDimensions} from '@libs/actions/connections/SageIntacct';
+import * as ErrorUtils from '@libs/ErrorUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@libs/Navigation/types';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import INPUT_IDS from '@src/types/form/SageIntacctDimensionsForm';
+import DimensionTypeSelector from './DimensionTypeSelector';
+
+type SageIntacctEditUserDimensionPageProps = StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.ACCOUNTING.SAGE_INTACCT_EDIT_USER_DIMENSION>;
+
+function SageIntacctEditUserDimensionPage({route}: SageIntacctEditUserDimensionPageProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    const editedUserDimensionName: string = route.params.dimensionName;
+    const [policy] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY}${route.params.policyID ?? '-1'}`);
+    const policyID: string = policy?.id ?? '-1';
+    const userDimensions = policy?.connections?.intacct?.config?.mappings?.dimensions;
+    const editedUserDimension = userDimensions?.find((userDimension) => userDimension.name === editedUserDimensionName);
+    const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+
+    const validate = useCallback(
+        (values: FormOnyxValues<typeof ONYXKEYS.FORMS.SAGE_INTACCT_DIMENSION_TYPE_FORM>) => {
+            const errors: FormInputErrors<typeof ONYXKEYS.FORMS.SAGE_INTACCT_DIMENSION_TYPE_FORM> = {};
+
+            if (!values[INPUT_IDS.INTEGRATION_NAME]) {
+                ErrorUtils.addErrorMessage(errors, INPUT_IDS.INTEGRATION_NAME, translate('common.error.fieldRequired'));
+            }
+
+            if (userDimensions?.some((userDimension) => userDimension.name === values[INPUT_IDS.INTEGRATION_NAME] && editedUserDimensionName !== values[INPUT_IDS.INTEGRATION_NAME])) {
+                ErrorUtils.addErrorMessage(errors, INPUT_IDS.INTEGRATION_NAME, translate('workspace.intacct.dimensionExists'));
+            }
+
+            if (!values[INPUT_IDS.DIMENSION_TYPE]) {
+                ErrorUtils.addErrorMessage(errors, INPUT_IDS.DIMENSION_TYPE, translate('common.error.fieldRequired'));
+            }
+            return errors;
+        },
+        [editedUserDimensionName, translate, userDimensions],
+    );
```

Comment:
Extracting this to a separate function seems kind of awkward because it needs data from policy and a translate function, so I would either need to create a hook, which seems to be an overkill, or I would need to pass a ton of weirdly typed params, which looks ugly. Since we have these functions repeated only in two places, maybe let's settle on what we have? WDYT

Reply:
right, hook is an overkill and passing all those translations is not too good too. Let's stick to what we have 👍